src/NHSD.BuyingCatalogue.Organisations.Api/Controllers/OdsController.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using Microsoft.AspNetCore.Mvc;
+using NHSD.BuyingCatalogue.Organisations.Api.ViewModels.Organisations;
+
+namespace NHSD.BuyingCatalogue.Organisations.Api.Controllers
+{
+    //[Authorize(Policy = Policy.CanAccessOrganisation)]
+    [Route("api/v1/ods")]
+    [ApiController]
+    [Produces("application/json")]
+    public sealed class OdsController : Controller
+    {
+        [HttpGet]
+        [Route("{odsCode}")]
+        public ActionResult GetByOdsCodeAsync(string odsCode)
+        {
+            if (odsCode is null)
+                throw new ArgumentNullException(nameof(odsCode));
+
+            if (string.IsNullOrWhiteSpace(odsCode))
```

Comment:
There is a case here for rather than returning an exception, the code could return a 404 notfound status instead. The code will eventually communicate with an external system, there maybe times when an empty or whitespace is a valid value. Unlikely in this instance but it will be up to the "repository" to decide that. This I believe would also be consistent with our other GET endpoints.

Reply:
Interesting point you're making. 

I agree, a 404 would be more appropriate here.